### DISCUSSION TITLE
[Compilers]<bn256 pairing multiplication not pre compiled>

### DISCUSSION THREAD

**Discussion by linyoufa** [zkSync-Community-Hub/zksync-developers#207] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.21

### zksolc/zkvyper version?

1.3.16

### Explain the issue

```bash
1.The following method calls failed

   success := staticcall(sub(gas(), 2000000), 6, mIn, 128, pR, 64) 
   success := staticcall(sub(gas(), 2000), 7, mIn, 96, mIn, 64)
   success := staticcall(sub(gas(), 2000), 8, _pPairing, 768, _pPairing, 0x20)

2.this is pre-compiled Ethereum:
 
var PrecompiledContractsIstanbul = map[common.Address]PrecompiledContract{
	common.BytesToAddress([]byte{1}): &ecrecover{},
	common.BytesToAddress([]byte{2}): &sha256hash{},
	common.BytesToAddress([]byte{3}): &ripemd160hash{},
	common.BytesToAddress([]byte{4}): &dataCopy{},
	common.BytesToAddress([]byte{5}): &bigModExp{eip2565: false},
	common.BytesToAddress([]byte{6}): &bn256AddIstanbul{},
	common.BytesToAddress([]byte{7}): &bn256ScalarMulIstanbul{},
	common.BytesToAddress([]byte{8}): &bn256PairingIstanbul{},
	common.BytesToAddress([]byte{9}): &blake2F{},
}

3.zksync precompiles
 Ecrecover.yul  Keccake256.yul  SHA256.yul
```

**Comment by Romsters** [comment, marked as answer]
Hi @linyoufa, these precompiles are not yet available, zksync team is actively working on them and they will be announced soon.

**Reply by linyoufa** [reply]
Thanks and stay tuned

**Reply by bxpana** [reply]
We should have support for this in the coming weeks. Keep an eye out on https://twitter.com/zkSyncDevs for updates

**Comment by dougtende** [comment]
It seems like you're encountering issues with calling precompiled contracts in zkSync using staticcall. Ensure that the gas provided for the staticcall is sufficient for the operation and that the correct precompiled contract address is being used. Additionally, verify that the parameters passed to the precompiled contracts are correct. If the issue persists, consider reaching out to the zkSync community or consulting the zkSync documentation for further assistance.

**Comment by robbmoser19** [comment]
The issue you're encountering seems to be related to failed method calls to precompiled contracts in zkSync. 
 Please ensure that the method calls are correctly formatted and that the precompiled contracts are properly initialized.
 Additionally, ensure that the gas limit provided for the method calls is sufficient for their execution.

You may also want to verify that the addresses of the precompiled contracts in zkSync match those defined in the Ethereum precompiled contracts for Istanbul.

 For further assistance, please review the zkSync documentation or seek support from their community.

 Repo Link: <insert your repo link here>

**Comment by springsj** [comment]
The issue may be due to the gas limit set for the staticcall operations being insufficient. Ensure that an appropriate amount of gas is allocated for each staticcall to prevent failures.

**Comment by inutpediflinda** [comment]
1. The method calls using `staticcall` with gas subtraction failed.
2. These are pre-compiled contracts for Ethereum Istanbul.
3. Zksync precompiles include Ecrecover, Keccake256, and SHA256.

**Comment by rafordtetseyd** [comment]
The issue is related to failed method calls in pre-compiled Ethereum contracts using zksolc/zkvyper version 1.3.16 with Solidity version 0.8.21. The specific method calls that failed are:

staticcall(sub(gas(), 2000000), 6, mIn, 128, pR, 64)
staticcall(sub(gas(), 2000), 7, mIn, 96, mIn, 64)
staticcall(sub(gas(), 2000), 8, _pPairing, 768, _pPairing, 0x20)
It is also noted that zksync has its own precompiled contracts for Ecrecover, Keccak256, and SHA256. The provided information does not include a link to the repository.

**Comment by eddiedenna** [comment]
The user is encountering issues with several method calls using staticcall in zksolc/zkvyper version 1.3.16 with Solidity version 0.8.21. They also mention pre-compiled Ethereum contracts and zksync precompiles. The user does not provide a link to their repo.
